### PR TITLE
changedetection-io: init at 0.39.20.4

### PR DESCRIPTION
**Pull request by SuperSandro2000**
Closes #184511

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
I am planning to merge this soonish. playwright already got a lot of review in the linked PR and the service is coming in another PR.

**Comment by SuperSandro2000** [comment]
Finally got everything working 🎉 

image

File: nixos/modules/services/web-apps/changedetection-io.nix [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    systemd = let
+      defaultStateDir = cfg.datastorePath == "/var/lib/changedetection-io";
+    in {
+      services.changedetection-io = {
+        wantedBy = [ "mutli-user.target" ];
+        after = [ "network.target" ];
+        preStart = ''
+          mkdir -p ${cfg.datastorePath}
+        '';
+        serviceConfig = {
```
Comment (yrd) [inline nixos/modules/services/web-apps/changedetection-io.nix:135]: Does this not work with `ProtectSystem` and `ProtectHome`?

**Review by yrd — APPROVED** [review]
Nice, thanks so much for finishing this up @SuperSandro2000 !

File: pkgs/development/python-modules/playwright/default.nix [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    # See here for the Chrome options:
+    # https://github.com/NixOS/nixpkgs/issues/136207#issuecomment-908637738
+    makeWrapper ${chromium}/bin/chromium $out/chromium-$CHROMIUM_REVISION/chrome-linux/chrome \
+      --set SSL_CERT_FILE /etc/ssl/certs/ca-bundle.crt \
+      --set FONTCONFIG_FILE ${fontconfig}
+  '' + lib.optionalString withFirefox ''
+    FIREFOX_REVISION=$(jq -r '.browsers[] | select(.name == "firefox").revision' $BROWSERS_JSON)
+    mkdir -p $out/firefox-$FIREFOX_REVISION
+    ln -s ${firefox}/bin/firefox $out/firefox-$FIREFOX_REVISION/firefox
+  '' + ''
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/python-modules/playwright/default.nix:131]: No, I didn't but I assume we probably need one or two iterations to fully get this working.

File: nixos/modules/services/web-apps/changedetection-io.nix (hunk 1 above)
Comment (SuperSandro2000) [reply nixos/modules/services/web-apps/changedetection-io.nix:135]: Added and seems to work.

File: pkgs/development/python-modules/playwright/default.nix [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  format = "setuptools";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "playwright-python";
+    rev = "v${version}";
+    # if setuptools_scm is not listing files via git almost all python files are excluded
+    leaveDotGit = true;
+    sha256 = "sha256-p7LJS8b9DQt79w+HpiHdSxNEHN6zwFmJzA7kyqkZ1C4=";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/playwright/default.nix:149]:
```suggestion
    sha256 = "sha256-KF5+dTNr3s9oeshSCkrim/WLWyPomKV1lPnLj4N+jb8=";
```

**Comment by SuperSandro2000** [comment]
TODO:
- remove leaveDotGit from playwright and setup the git repo manually

File: pkgs/development/python-modules/playwright/default.nix [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+  pname = "playwright";
+  version = "1.27.1";
+  format = "setuptools";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "playwright-python";
+    rev = "v${version}";
+    # if setuptools_scm is not listing files via git almost all python files are excluded
```
Comment (yrd) [inline pkgs/development/python-modules/playwright/default.nix:147]: Would you mind elaborating here? I tested this quickly and `SETUPTOOLS_SCM_PRETEND_VERSION = version;` seems to work...

**Comment by SuperSandro2000** [comment]
TODO: fix playwright src hash that keeps changing

File: pkgs/development/python-modules/playwright/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/playwright/default.nix:147]: It misses a lot files from https://github.com/microsoft/playwright-python/tree/main/playwright which are important for changedetection-io to work. I don't fully understand why and lack the knowledge of setuptools-scm to work around this.

**Pull request merged**